### PR TITLE
HueyoService is now a foreground service with a persistent notification

### PR DESCRIPTION
**Pull request by moubarak**
[bounty](https://assembly.com/hueyo/bounties/2)

### CODE REVIEW

**Comment by kurtisnelson** [comment]
Why is reflection needed?

**Comment by moubarak** [comment]
this hack is to make foreground services work prior to Android [API level 5](http://developer.android.com/guide/topics/manifest/uses-sdk-element.html). i just saw that [Hueyo requires Android 4.4 and up](https://play.google.com/store/apps/details?id=com.thisisnotajoke.hueyo) so that hack could be replaced with a simple call to setForeground(). If you want to do so just let me know.

from [Service class docs](http://developer.android.com/reference/android/app/Service.html)

> If you need your application to run on platform versions prior to API level 5, you can use the following model to call the older setForeground()

**Comment by kurtisnelson** [comment]
Yeah, all versions of Android actually still out there no longer require reflection.

**Comment by moubarak** [comment]
Ok removed reflection and updated deprecated methods.

File: app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java [hunk 1]
```diff
@@ -58,10 +62,11 @@ HueyoService getService() {
             return HueyoService.this;
         }
     }
-
```
Comment (kurtisnelson) [inline app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java:61]: Can we kill this whitespace?

Comment (moubarak) [inline app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java:61]: Which?

Comment (moubarak) [inline app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java:61]: i think i did

File: app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java [hunk 2]
```diff
@@ -38,8 +41,9 @@
 
 public class HueyoService extends Service {
     private static final String TAG = "HueyoService";
+	private static final int ONGOING_NOTIFICATION_ID = 101;
     private final IBinder mBinder = new LocalBinder();
-
+
```
Comment (kurtisnelson) [inline app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java]: The trailing whitespace.

Comment (moubarak) [inline app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java]: That was a tab. it didn't appear in Studio just here. Anyway replaced the tab with whitespace.

Comment (moubarak) [inline app/src/main/java/com/thisisnotajoke/hueyo/controller/HueyoService.java]: i just did one instance ill replace tabs with spaces in my settings. i don't see any trailing spaces in my editor.

**Comment by moubarak** [comment]
Ok converted those tab indents to spaces

**Pull request merged**